include/openssl/base.h

```diff
@@ -182,6 +182,22 @@ extern "C" {
 
 #endif
 
+#if defined(AWSLC_DEBUG_BUILD)
```

Comment:
Were we to automatically detect debug/release builds (see other comment), I think we should still have an explicit `AWSLC_DEBUG_BUILD` macro for those that would like it for their release builds.
```C++
#if defined(DEBUG) || defined(_DEBUG) || !defined(NDEBUG)
   #define AWSLC_DEBUG_BUILD 1
#endif

#if defined(AWSLC_DEBUG_BUILD)
...
```